sql/api/src/main/scala/org/apache/spark/sql/catalyst/parser/DataTypeAstBuilder.scala

```diff
@@ -138,7 +165,7 @@ class DataTypeAstBuilder extends SqlBaseParserBaseVisitor[AnyRef] {
       }
     } else {
       val badType = typeCtx.unsupportedType.getText
-      val params = typeCtx.INTEGER_VALUE().asScala.toList
+      val params = typeCtx.integerValue().asScala.map(_.getText).toList
```

Comment:
So the assumption here is the parameter markers should have been replaces and we can safely get the original INTEGER_VALUE text?